core/src/main/java/org/apache/iceberg/ManifestFiles.java

```diff
@@ -81,7 +81,7 @@ static ContentCache contentCache(FileIO io) {
   }
 
   /** Drop manifest file cache object for a FileIO if exists. */
-  public static synchronized void dropCache(FileIO fileIO) {
+  public static void dropCache(FileIO fileIO) {
```

Comment:
@findepi Thanks for pinging me, I'm not the author here, so I was hoping that she/he would chime in here. I'm assuming here that this is synchronized because the `.invalidate` can throw a NPE in case of a race condition when two threads try to remove the same `fileIO` object:
![image](https://github.com/apache/iceberg/assets/1134248/b6cab1e9-8866-412f-b117-dc84b7718e0d)

Reply:
I shouldn't comment before my morning coffee. This is about the key that's being passed in.